src/main/java/at/hannibal2/skyhanni/features/rift/area/dreadfarm/RiftWiltedBerberisHelper.kt

```diff
@@ -145,12 +237,36 @@ object RiftWiltedBerberisHelper {
         }
     }
 
-    @HandleEvent
-    fun onConfigFix(event: ConfigUpdaterMigrator.ConfigFixEvent) {
-        event.move(60, "rift.area.dreadfarm.wiltedBerberis.hideparticles", "rift.area.dreadfarm.wiltedBerberis.hideParticles")
+    private fun primaryRender(event: LorenzRenderWorldEvent) {
+        if (berberisList.isEmpty()) return
+        var alpha = 0.8f
+        var previousBerberis: LorenzVec? = null
+        event.drawDynamicText(berberisList[0].up(), "§eWilted Berberis", 1.5, ignoreBlocks = false)
+
+        berberisList.take(config.previewCount + 1).forEachIndexed { i, loc ->
+            // box it with half the opacity of the previous box, first in list is highlighted
+            if (i == 0) event.drawFilledBoundingBoxNea(axisAlignedBB(loc), config.highlightColor.toSpecialColor(), alpha)
+            else event.drawFilledBoundingBoxNea(axisAlignedBB(loc), Color.WHITE, alpha)
+            alpha *= 0.6f
+
+            // if there's a previous berberis, draw a line to it. The line from the 2nd to the 1st should be highlighted
+            if (i == 1) {
+                previousBerberis?.let {
+                    event.draw3DLine(loc.add(0.5, 0.5, 0.5), it.add(0.5, 0.5, 0.5), config.highlightColor.toSpecialColor(), 4, false)
+                }
+            } else {
+                previousBerberis?.let {
+                    event.draw3DLine(loc.add(0.5, 0.5, 0.5), it.add(0.5, 0.5, 0.5), Color.WHITE, 2, false)
+                }
+            }
+
+            previousBerberis = loc
+        }
     }
 
-    private fun axisAlignedBB(loc: LorenzVec) = loc.add(0.1, -0.1, 0.1).boundingToOffset(0.8, 1.0, 0.8).expandBlock()
+    private fun nearestBerberis(location: LorenzVec): WiltedBerberis? =
```

Comment:
try to change the logic so that the distance is only getting calculated once per entry, not twice